### DISCUSSION TITLE
Sobre consumo de .dlls e engasgo de UI.

### DISCUSSION THREAD

**Discussion by victorruandev** [Flutterando/forum#39] [Perguntas e Respostas]
Olá! Espero estar utilizando bem o fórum! 

Então, a questão que venho trazer é sobre o consumo de dlls. Sim, já vi o vídeo da Flutterando onde o Jacob fala sobre _Dart ffi_       
( https://www.youtube.com/watch?v=RF5sPE4ootk&t=2861s), mas infelizmente, e não sendo mesmo a proposta do vídeo, não resolve meu problema. Ele gira em  torno do consumo de uma DLL (Windows) que trava (provavelmente o _Event Loop_ fica engasgando com a task de rodar a função da dll) a UI e depois perde a conexão (_Lost connection to device._). Estou preso nessa tentativa de solução já faz quase um mês e venho aqui pra tentar solucionar meu caso. Aceito susgestões e qualquer tipo de ajudar. Vou estar sempre verificando aqui para ver o que vocês tem a falar, mas vou deixar meu email: victorruan135@gmail.com.

Valeu pessoal!

**Comment by eduardoflorence** [comment]
@victorruandev para este cenário você tem que usar o recurso de Isolate do Dart para ele consumir sua DLL num “isolado” separado do principal e assim não travar a UI. Dá uma lida neste artigo para entender melhor como usar o Isolate:
https://blog.logrocket.com/multithreading-flutter-using-dart-isolates

**Reply by jacobaraujo7** [reply]
Infelizmente coisas que usam FFI por enquanto não funcionam em um Isolate.
Terá que fazer a concorrência no C/C++ abrindo uma thread.

**Reply by eduardoflorence** [reply]
Oi @jacobaraujo7, você acredita que a solução dada no link abaixo seja um possível “hack” para permitir usar o isolate com o dart FFI?
https://stackoverflow.com/a/67920994

**Reply by victorruandev** [reply]
Olá @eduardoflorence, eu já vinha tentando utilizar Isolates mas infelizmente não o consegui :(. Tentarei o "hack" que vc sugestionou!

**Reply by victorruandev** [reply]
@jacobaraujo7 obrigado pela informação, irei pesquisar sobre a abertura dessa thread para tentar solucionar o problema!

**Reply by jacobaraujo7** [reply]
> Oi @jacobaraujo7, você acredita que a solução dada no link abaixo seja um possível “hack” para permitir usar o isolate com o dart FFI?
> 
> https://stackoverflow.com/a/67920994

Isso mesmo.
Quando instalamos o dart, vem uma pasta runtime que contém os headers para comunicação por mensageria. Seria a mesma ideia do MethodChannel.

**Comment by victorruandev** [comment]
@eduardoflorence e @jacobaraujo7 , estou tentando compreender aqui a proposta de solução enviada, mas não estou conseguindo abstrair! Vocês podem indicar algum conteúdo onde eu possar me apoiar!? 
Desde já muito agradecido pela ajuda até aqui!!!

**Comment by victorruandev** [comment]
Estou pensando em fazer uma API (Dart) e servir o app a partir desse serviço. Acham que seria uma boa possibilidade? Pois estou encontrando muita complexidade para criar a thread no C/C++ e dar fazer a conexão com o Dart!
